test/nn/test_model_summary.py

```diff
@@ -0,0 +1,196 @@
+        '| GCN                 | [100, 32], [100, 100] | [100, 32]      | 1,072    |',  # noqa
+        '| ├─(act)ReLU         | [100, 16]             | [100, 16]      | --       |',  # noqa
+        '| ├─(convs)ModuleList | --                    | --             | 1,072    |',  # noqa
+        '| │    └─(0)GCNConv   | [100, 32], [100, 100] | [100, 16]      | 528      |',  # noqa
+        '| │    └─(1)GCNConv   | [100, 16], [100, 100] | [100, 32]      | 544      |',  # noqa
+        '+---------------------+-----------------------+----------------+----------+',  # noqa
+    ])
+
+    assert summary(model, x, adj_t) == expected
+
+    expected = '\n'.join([
+        '+---------------------+--------------------+----------------+----------+',  # noqa
+        '| Layer               | Input Shape        | Output Shape   | #Param   |',  # noqa
+        '|---------------------+--------------------+----------------+----------|',  # noqa
+        '| GCN                 | [100, 32], [2, 20] | [100, 32]      | 1,072    |',  # noqa
+        '| ├─(act)ReLU         | [100, 16]          | [100, 16]      | --       |',  # noqa
+        '| ├─(convs)ModuleList | --                 | --             | 1,072    |',  # noqa
+        '+---------------------+--------------------+----------------+----------+',  # noqa
+    ])
+
+    assert summary(model, x, edge_index, max_depth=1) == expected
+
+    expected = '\n'.join([
+        '+-----------------------------------------+--------------------+----------------+----------+',  # noqa
+        '| Layer                                   | Input Shape        | Output Shape   | #Param   |',  # noqa
+        '|-----------------------------------------+--------------------+----------------+----------|',  # noqa
+        '| GCN                                     | [100, 32], [2, 20] | [100, 32]      | 1,072    |',  # noqa
+        '| ├─(act)ReLU                             | [100, 16]          | [100, 16]      | --       |',  # noqa
+        '| ├─(convs)ModuleList                     | --                 | --             | 1,072    |',  # noqa
+        '| │    └─(0)GCNConv                       | [100, 32], [2, 20] | [100, 16]      | 528      |',  # noqa
+        '| │    │    └─(aggr_module)SumAggregation | [120, 16], [120]   | [100, 16]      | --       |',  # noqa
+        '| │    │    └─(lin)Linear                 | [100, 32]          | [100, 16]      | 512      |',  # noqa
+        '| │    └─(1)GCNConv                       | [100, 16], [2, 20] | [100, 32]      | 544      |',  # noqa
+        '| │    │    └─(aggr_module)SumAggregation | [120, 32], [120]   | [100, 32]      | --       |',  # noqa
+        '| │    │    └─(lin)Linear                 | [100, 16]          | [100, 32]      | 512      |',  # noqa
+        '+-----------------------------------------+--------------------+----------------+----------+',  # noqa
+    ])
+
+    assert summary(model, x, edge_index, leaf_module=None) == expected
+
```

Comment:
Neat! I would recommend having only one assert per test (split test_basic_summary into multiple functions and use that fixture to avoid re-computation). It is easier to figure out what is wrong when a test fails, and if there is more than one failed test, we can see all of them at once.

Reply:
Sounds great! Will update tests soon. Thanks for your helpful suggestions!